### PR TITLE
Run gix-odb tests under SHA-256

### PR DESCRIPTION
**Pull request by 10ne1**
**What**

`gix-odb` handles SHA-256 fine, but its tests opened fixtures with the SHA-1-defaulting `gix_odb::at()`, so the SHA-256 paths went untested, so we make the dynamic-store, memory and loose-store tests hash-aware and we add the SHA-256 test runs to the CI.

`gix-odb` was the last crate in the object-storage stack without these runs.

Part of #281.
 
 **Testing**
 
 Full `gix-odb` suite passes under both hashes, including the `parallel`-gated test:
 ```
GIX_TEST_FIXTURE_HASH=sha1   cargo nextest run -p gix-odb --features parallel
GIX_TEST_FIXTURE_HASH=sha256 cargo nextest run -p gix-odb --features parallel
```

**AI disclosure**

This PR was written with the help of Claude Code, but was **not** vibe-coded. I'm not a fan of vibe-coding. I used AI to better understand the codebase myself (asked it a lot of questions), review both my code and the code generated by AI, used it especially to detect bugs and corner-cases and suggest fixes, however I very carefully reviewed & edited each code/comment lines, the commit messages and so on, until I was satisfied with the result.

### CODE REVIEW

File: gix-odb/tests/odb/main.rs [hunk 1]
```diff
@@ -4,11 +4,35 @@ use gix_testtools::fixture_path;
 pub use gix_testtools::{scripted_fixture_read_only, scripted_fixture_read_only_with_args, scripted_fixture_writable};
 
 pub fn hex_to_id(hex: &str) -> ObjectId {
-    ObjectId::from_hex(hex.as_bytes()).expect("40 bytes hex")
+    ObjectId::from_hex(hex.as_bytes()).expect("valid hex of the configured hash length")
+}
```
[the remark this answers is not in the document]
Comment (10ne1) [reply gix-odb/tests/odb/main.rs:8]: Updated the msg to use Codex's wording.

**Review by Byron — COMMENTED** [review]
Thanks a lot, much appreciated.

Please note that I rewrote your commits partially and decided to remove the big fixtures that already existed in similar sizes for SHA-1. Unfortunately there was no way around that except for taking you into another review round and this is something I will always try to avoid.

Nothing else changed though, as I put my review changes into a new commit accordingly.

Also, for some reason, I still couldn't push using the normal GitHub mechanism. So I accepted your invite to collaborate on your fork, which fortunately works :).

File: gix-odb/tests/fixtures/generated-archives/.gitignore [hunk 2]
```diff
@@ -5,7 +5,18 @@
 # `make_alternates_odb.sh` writes `$PWD/object_source/.git/objects` into
 # `.git/objects/info/alternates`; the absolute path is host-specific.
 make_alternates_odb.tar
+# The fixuture is 2MB in size, so let's not track it (like its SHA-1 sibling)
+make_alternates_odb_sha256.tar
```
[the remark this answers is not in the document]
Comment (Byron) [reply gix-odb/tests/fixtures/generated-archives/.gitignore:9]: my bad 😅

File: gix-odb/tests/fixtures/generated-archives/.gitignore [hunk 3]
```diff
@@ -5,7 +5,18 @@
...
 repo_with_loose_objects.tar
+repo_with_loose_objects_sha256.tar
+# The fixuture is 2MB in size, so let's not track it (like its SHA-1 sibling)
+make_repo_multi_index_sha256.tar
+# The fixuture is 2MB in size, so let's not track it (like its SHA-1 sibling)
+make_replaced_history_sha256.tar
+# The fixuture is 2MB in size, so let's not track it (like its SHA-1 sibling)
+make_repo_multi_index_no_sha256.tar
+# The fixuture is 2MB in size, so let's not track it (like its SHA-1 sibling)
+make_repo_multi_index_without-multi-index_sha256.tar
```
[the remark this answers is not in the document]
Comment (Byron) [reply gix-odb/tests/fixtures/generated-archives/.gitignore:22]: my bad 😅

File: gix-odb/tests/fixtures/generated-archives/.gitignore [hunk 4]
```diff
@@ -5,7 +5,18 @@
...
+make_alternates_odb_sha256.tar
 # Fixture deliberately uses loose objects; whether commits remain loose vs. get
 # packed depends on installation-wide Git auto-gc/repack behavior, so the
 # layout is regenerated per run.
 repo_with_loose_objects.tar
+repo_with_loose_objects_sha256.tar
+# The fixuture is 2MB in size, so let's not track it (like its SHA-1 sibling)
+make_repo_multi_index_sha256.tar
+# The fixuture is 2MB in size, so let's not track it (like its SHA-1 sibling)
+make_replaced_history_sha256.tar
```
[the remark this answers is not in the document]
Comment (Byron) [reply gix-odb/tests/fixtures/generated-archives/.gitignore:18]: my bad 😅

File: gix-odb/tests/fixtures/generated-archives/.gitignore [hunk 5]
```diff
@@ -5,7 +5,15 @@
...
 # Fixture deliberately uses loose objects; whether commits remain loose vs. get
 # packed depends on installation-wide Git auto-gc/repack behavior, so the
 # layout is regenerated per run.
 repo_with_loose_objects.tar
+repo_with_loose_objects_sha256.tar
+# The fixuture is 2MB in size, so let's not track it (like its SHA-1 sibling)
+make_repo_multi_index_sha256.tar
+# The fixuture is 2MB in size, so let's not track it (like its SHA-1 sibling)
+make_repo_multi_index_no_sha256.tar
+# The fixuture is 2MB in size, so let's not track it (like its SHA-1 sibling)
```
[the remark this answers is not in the document]
Comment (Byron) [reply gix-odb/tests/fixtures/generated-archives/.gitignore:21]: my bad 😅
